### PR TITLE
Redirect robots.txt if hosting web content

### PR DESCRIPTION
**Pull request by crobibero**
Fixes https://github.com/jellyfin/jellyfin/issues/4689

Redirects `{baseUrl}/robots.txt` to `{baseUrl}/web/robots.txt`

### CODE REVIEW

**Comment by joshuaboniface** [comment]
As mentioned on Matrix, I'd consider it a blocker that we broke `/emby` API endpoints in this release. I think it would be ideal to generalize this from `RobotsRedirectionMiddleware` to just `RedirectionMiddleware` and add a redirection for any `/emby` API endpoints to `/` so those continue to work. Thoughts?

**Comment by nielsvanvelzen** [comment]
The /emby redirection should happen internally because clients might not follow redirects.

**Comment by crobibero** [comment]
`/emby` and `/mediabrowser` route rewrite has been added, and tested with and without a baseurl set.

**Comment by cvium** [comment]
> The /emby redirection should happen internally because clients might not follow redirects.

I have the same concern

**Comment by crobibero** [comment]
> 
> 
> > The /emby redirection should happen internally because clients might not follow redirects.
> 
> I have the same concern

I opted to not redirect, but to just rewrite the route internally

**Comment by joshuaboniface** [comment]
Great, that's how I was thinking too - completely transparent to the client(s), just treat `/emby/users` and `/users` as the exact same basically.

**Review by joshuaboniface — APPROVED** [review]

File: Jellyfin.Server/Middleware/PathTrimMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Server.Middleware
+{
+    /// <summary>
+    /// Removes /emby and /mediabrowser from requested route.
+    /// </summary>
+    public class PathTrimMiddleware
```
Comment (cvium) [inline Jellyfin.Server/Middleware/PathTrimMiddleware.cs:11]: Can we call it something more descriptive? Like `LegacyEmbyRouteRewrite` or something.

**Review by oddstr13 — APPROVED** [review]

**Pull request merged**